src/main/java/com/swervedrivespecialties/exampleswerve/util/BeakXboxController.java

```diff
@@ -26,35 +63,91 @@
 
     public BeakXboxController(int port){
         controller = new XboxController(port);
-        a = new JoystickButton(controller,RobotMap.BUTTON_A);
-        b = new JoystickButton(controller,RobotMap.BUTTON_B);
-        x = new JoystickButton(controller,RobotMap.BUTTON_X);
-        y = new JoystickButton(controller,RobotMap.BUTTON_Y);
-        LEFT_BUMPER = new JoystickButton(controller,RobotMap.BUTTON_LB);
-        RIGHT_BUMPER = new JoystickButton(controller,RobotMap.BUTTON_RB);
-        BACK = new JoystickButton(controller,RobotMap.BUTTON_BACK);
-        START = new JoystickButton(controller,RobotMap.BUTTON_START);
-        LEFT_TRIGGER = new JoystickButton(controller,RobotMap.LEFT_TRIGGER); //true
-        RIGHT_TRIGGER = new JoystickButton(controller,RobotMap.RIGHT_TRIGGER); //false
-    //still need to create dpad directions
+        a = new JoystickButton(controller, BUTTON_A_INDEX);
+        b = new JoystickButton(controller, BUTTON_B_INDEX);
+        x = new JoystickButton(controller, BUTTON_X_INDEX);
+        y = new JoystickButton(controller, BUTTON_Y_INDEX);
+        LEFT_BUMPER = new JoystickButton(controller, BUTTON_LB_INDEX);
+        RIGHT_BUMPER = new JoystickButton(controller,BUTTON_RB_INDEX);
+        BACK = new JoystickButton(controller, BUTTON_BACK_INDEX);
+        START = new JoystickButton(controller, BUTTON_START_INDEX);
+        //true = left, false = right (right now)
+        LEFT_TRIGGER = new TriggerButton(controller, LEFT_TRIGGER_BOOLEAN); 
+        RIGHT_TRIGGER = new TriggerButton(controller, RIGHT_TRIGGER_BOOLEAN); 
+        DPAD_UP = new DpadButton(controller, Directions.UP);
+        DPAD_DOWN = new DpadButton(controller, Directions.DOWN);
+        DPAD_LEFT = new DpadButton(controller, Directions.LEFT);
+        DPAD_RIGHT = new DpadButton(controller, Directions.RIGHT);
     }
+
     public double getLeftXAxis(){
-        return controller.getRawAxis(0);
+        return controller.getRawAxis(LEFT_X_AXIS_INDEX);
     }
+
     public double getLeftYAxis(){
-        return controller.getRawAxis(1);
+        return controller.getRawAxis(LEFT_Y_AXIS_INDEX);
     }
+
     public double getLeftTrigger(){
-        return controller.getRawAxis(2);
+        return controller.getRawAxis(LEFT_TRIGGER_AXIS_INDEX);
     }
+
     public double getRightTrigger(){
-        return controller.getRawAxis(3);
+        return controller.getRawAxis(RIGHT_TRIGGER_AXIS_INDEX);
     }
+
     public double getRightXAxis(){
-        return controller.getRawAxis(4);
+        return controller.getRawAxis(RIGHT_X_AXIS_INDEX);
     }
     public double getRightYAxis(){
-        return controller.getRawAxis(5);
+        return controller.getRawAxis(RIGHT_Y_AXIS_INDEX);
+    }
+
+    public static class TriggerButton extends Button {
```

Comment:
The way you wrote this had a bug. It called the controller in the constructor of the class. Take a look at it, and if you don't see the bug, call me and I can talk you through it.

Reply:
The hand thing perhaps isn't ideal, the best way to do this would actually probably be to use the RawAxis constant from above, but this was just a quick fix. you could add TODO: fix it to use the constant from above and a raw axis call, just for stylistic completeness, but I find it extremely unlikely this will cause a bug in the foreseeable future.